docs/core/tools/dotnet-environment-variables.md

```diff
@@ -183,6 +183,22 @@ See [EventPipe environment variables](../diagnostics/eventpipe.md#trace-using-en
 
 Specifies the location of the .NET runtimes, if they are not installed in the default location. The default location on Windows is `C:\Program Files\dotnet`. The default location on Linux and macOS is `/usr/local/share/dotnet`. This environment variable is used only when running apps via generated executables (apphosts). `DOTNET_ROOT(x86)` is used instead when running a 32-bit executable on a 64-bit OS.
 
+### `DOTNET_HOST_PATH`
+
+Specifies the absolute path to a `dotnet` binary (`dotnet.exe` on Windows, `dotnet` on non-Windows) that was used to launch this session. This is typically used by a hosting application that wants to ensure that the exact same version of the host binary is used for all child processes.
+
+Without this environment variable, if a `dotnet` binary was used from a location not on the `PATH`, child processes of that `dotnet` that wished to invoke `dotnet` would invoke a different `dotnet` binary than the one that was used to spawn them - this can result in confusing behavior for end users.
+
+As a tool author, if you want to locate the correct `dotnet` binary to use to spawn a .NET binary, you should use the following algorithm:
+
+* if `DOTNET_HOST_PATH` is set, use that value directly
+* otherwise, probe `PATH` according to the rules of your operating system to find the first `dotnet` binary on the `PATH`
+
+This environment variable is set by default when the .NET SDK is invoked (typically by using `dotnet build`, `dotnet publish`, and so on). If your application needs to communicate the location of the `dotnet` binary that ran it in a similar way to the SDK, you should also set this environment variable.
+
+> [!NOTE]
+> This environment variable will *not* be set for apphost-based execution (applications that are directly launched via a binary like `myapp.exe` or `myapp`) or through applications that are run via the `dotnet` binary (like `dotnet myapp.dll`). In such cases, the spawned applications should locate `dotnet` via the `PATH` as described above.
```

Comment:
Eventually we should also describe the recommended setup for running "apps" as child processes. Outside of SDK we're pushing people to use executables as the app's entry point, and to avoid the `dotnet app.dll` gesture. (It is problematic for UI apps, and just doesn't work at all for single-file apps). So if we envision usage of executable entrypoitns in this chain then `DOTNET_HOST_PATH` is not going to solve anything.